txpool/txpool_test.go

```diff
@@ -4363,10 +4399,24 @@ func BenchmarkAddTxTime(b *testing.B) {
 			require.NoError(b, err, "fail to create pool")
 			pool.SetSigner(signer)
 
-			for i := 0; i < len(txs); i++ {
-				err = pool.addTx(local, txs[i])
-				require.NoError(b, err)
+			lenTxs := len(txs)
+			wg := sync.WaitGroup{}
+			wg.Add(lenTxs)
+
+			for i := 0; i < lenTxs; i++ {
+				tx := txs[i]
+
+				go func() {
+					defer wg.Done()
+
+					errAdd := pool.addTx(local, tx)
+					if err != nil {
```

Comment:
I am against this kind of testing, it is not really deterministic, since we are expecting no error, but if it happens it should be this kind of error. I don't think this test was expecting an error in the first place, but it expected that transaction would be replaced.

I suggest either changing the test so it does not return the `ReplacementUnderpriced`, or reverting it to old behavior.

Reply:
If the test stays this way, I believe this is an error that we need to check.

```suggestion
					if errAdd != nil {
```

Reply:
I agree with what was previously said. Maybe a compromise is to count the exact number of applied transactions after sync (transactions don't have repeated nonce). Of course, it would still have the ability to reject a parallel transaction with a worse price, which is expected and is a realistic picture for measuring this type of performance.